Automattic/newspack-newsletters
# fix: only show public newsletters on CPT archive pages

### All Submissions:






### Changes proposed in this Pull Request:



#398 introduced a regression on Newsletter CPT archives where all published/sent newsletters are shown, whether or not they are marked as public. This fixes the query on those archives so that only those newsletters marked as publicly visible will appear, as in category and tag archives.

Closes #403.

### How to test the changes in this Pull Request:

1. Create and send several newsletters, but mark only some of them as publicly visible.
2. On `master`, visit the `https://<your_site_url>/newsletter` archive page (or whatever the slug you've defined for newsletters in Newsletter settings). Observe that all sent newsletters are displayed here, even those not marked as publicly visible.
3. Check out this branch and refresh the archive page. Confirm that now only those newsletters marked as publicly visible are shown.
4. Change the "publicly visible" setting for some newsletters and confirm that the posts shown in the archive are updated accordingly.
5. Apply some tags and categories to your newsletters, both public and not. Visit the archive pages for those terms and confirm that you still see all published posts with the term, plus only the newsletters that are marked as publicly visible. (This behavior should be unchanged from `master`).

### Other information:
